pkg/openstack/loadbalancer.go

```diff
@@ -2317,6 +2344,14 @@ func (lbaas *LbaasV2) ensureAndUpdateOctaviaSecurityGroup(clusterName string, ap
 			return fmt.Errorf("failed to create Security Group for loadbalancer service %s/%s: %v", apiService.Namespace, apiService.Name, err)
 		}
 		lbSecGroupID = lbSecGroup.ID
+
+		tags := lbaas.getCustomLoadBalancerTags(apiService, svcConf)
+		for _, tag := range tags {
+			if err := neutrontags.Add(lbaas.network, "securitygroups", lbSecGroupID, tag).ExtractErr(); err != nil {
```

Comment:
Ouch, true. That might be a lot of calls, let's try to use `neutrontags.ReplaceAll()` instead. Something like this should do. Note that I've improved the error message a bit and replaced "securitygroups" with "security-groups" (fixing a previous mistake).

```
		tags := lbaas.getCustomLoadBalancerTags(apiService, svcConf)
		if _, err := neutrontags.ReplaceAll(lbaas.network, "security-groups", lbSecGroupID, neutrontags.ReplaceAllOpts{Tags: tags}).Extract(); err != nil {
			return fmt.Errorf("failed to add custom tags %s to security group %s (%s)", tags, lbSecGroupID, lbSecGroupName)
		}
```

Reply:
** An Opinion

i was looking into the codebase,this update all existing tags :
```
		tags := lbaas.getCustomLoadBalancerTags(apiService, svcConf)
		if _, err := neutrontags.ReplaceAll(lbaas.network, "security-groups", lbSecGroupID, neutrontags.ReplaceAllOpts{Tags: tags}).Extract(); err != nil {
			return fmt.Errorf("failed to add custom tags %s to security group %s (%s)", tags, lbSecGroupID, lbSecGroupName)
		}
```

this create a new tag to the resource and i think using the creation instead of the update is kinda valid because of it isb an O(n) and the little time difference won't be noticed I think but i will make the changes as requested
```
for _, tag := range tags {
		if err := neutrontags.Add(lbaas.network, "securitygroups", lbSecGroupID, tag).ExtractErr(); err != nil {
				return fmt.Errorf("failed to add custom tag %q to securitygroups neutrontags ", tag)
		}
}
```

Reply:
The time difference will be noticed - each of these `neutrontags.Add()` is a call to OpenStack Neutron API. In practice it can be really slow [^1] so saving calls to Neutron is really important.

[^1]: I've even created a meme once: https://dulek.github.io/kuryr-kubernetization/#18